### PR TITLE
New dashboard build

### PR DESCRIPTION
**Pull request by jonasrosland**
Simplifying the Dockerfile
Removing the HTML, JS and CSS minification
Renaming the team name according to our new standard and adding a link to our website
Added a few dependencies to make it build correctly

Signed-off-by: <jonas.rosland@gmail.com>

### CODE REVIEW

File: dashboard/Dockerfile [hunk 1]
```diff
@@ -13,13 +13,16 @@ RUN gem install compass
...
-  npm install &&\
-  bower install --allow-root --force-latest --config.interactive=false &&\
-  grunt build &&\
-  cp -r dist /app
+WORKDIR /opt
+RUN git clone https://github.com/codedellemc/mars-challenge
+
+WORKDIR /opt/mars-challenge/dashboard/
+
+RUN npm install
```
Comment (codenrhoden) [inline dashboard/Dockerfile]: Is it good or bad to split these into separate steps? I thought the advice was to combine what you could so you didn't create so many layers...

File: dashboard/Gruntfile.js [hunk 2]
```diff
@@ -335,10 +335,10 @@ module.exports = function (grunt) {
...
       dist: {
         options: {
-          collapseWhitespace: true,
-          conservativeCollapse: true,
-          collapseBooleanAttributes: true,
-          removeCommentsFromCDATA: true
+          collapseWhitespace: false,
+          conservativeCollapse: false,
+          collapseBooleanAttributes: false,
+          removeCommentsFromCDATA: false
```
Comment (codenrhoden) [inline dashboard/Gruntfile.js]: woohoo!

**Comment by codenrhoden** [comment]
Hey @jonasrosland 

I built this locally, and saw a couple weird things.  In order to build locally, I changed the Dockerfile to have:

```
-WORKDIR /opt
-RUN git clone https://github.com/codedellemc/mars-challenge
+#WORKDIR /opt
+#RUN git clone https://github.com/codedellemc/mars-challenge
+RUN mkdir -p /opt/mars-challenge/dashboard
+COPY . /opt/mars-challenge/dashboard
```

That seemed to build fine. But when I run the app, here is what I see:

screen shot 2016-11-30 at 9 52 33 am

Poking around a bit, it's missing a bunch of images in `/app/images`, but I don't know why:

```
root@234b8afcba5b:/app# tree
.
|-- 404.html
|-- favicon.ico
|-- images
|   |-- dead.0a6dd72b.png
|   `-- flare-on.635eb7a5.png
|-- index.html
|-- robots.txt
|-- scripts
|   |-- scripts.170cbe0a.js
|   `-- vendor.d7b43599.js
`-- styles
    |-- main.5e90a2ea.css
    `-- vendor.ec0d67b4.css

3 directories, 10 files
root@234b8afcba5b:/app# cd /opt/mars-challenge/dashboard/
root@234b8afcba5b:/opt/mars-challenge/dashboard# ls
Dockerfile    README.md  bower.json        dashboard_nginx.conf  node_modules  start.sh
Gruntfile.js  app        bower_components  dist                  package.json  test
root@234b8afcba5b:/opt/mars-challenge/dashboard# ls app/images/
background.jpg  dead.png  flare-off.png  flare-on.png  shield-off.png  shield-on.png  yeoman.png
```

As you can see, only two images made it in the compiled app, `flare-on.png` and `dead.png`. Any thoughts?

**Comment by codenrhoden** [comment]
I rebuilt everything a second time (disabling the Docker build cache), and the outcome was different. This time it only had the background image present. Nothing else!

**Comment by codenrhoden** [comment]
I was able to get things to work if I modified `Gruntfile.js`to no longer do image minification and just copied the images straight into `/images` instead.

I'm considering merging this, then doing a second commit to make that change.

Can you chime in, @jonasrosland ? Let me know if you saw this problem or anything? strange that you did not. That worries me.

File: dashboard/Dockerfile (hunk 1 above)
Comment (jonasrosland) [inline dashboard/Dockerfile]: Much easier to troubleshoot and you don't have to go through the 5 minute process every time if something breaks at the end.

File: dashboard/Gruntfile.js (hunk 2 above)
Comment (jonasrosland) [inline dashboard/Gruntfile.js]: :)

**Pull request merged**